liquidator/liquidator.js

```diff
@@ -158,12 +168,50 @@ class Liquidator {
       // Note: query the time again during each iteration to ensure the deadline is set reasonably.
       const currentBlockTime = this.empClient.getLastUpdateTime();
 
+      // Calculate the amount of tokens we will attempt to liquidate.
+      let tokensToLiquidate;
+
+      // If the user specifies `maxTokensToLiquidateWei`, then we must make sure that it follows an important constraint:
+      // we cannot bring the position down below the `minSponsorPosition`.
+      if (maxTokensToLiquidateWei) {
```

Comment:
@mrice32 Tried to break this calculation down

Reply:
ye, this is great. thanks for adding all these comments and splitting up this logic